### PR TITLE
[ChainOps] k8s Productionization 

### PR DESCRIPTION
**Pull request by intl-man**
This should now allow us to run a multi-validator network on k8s. Several new rake commands have been introduced, in an attempt to make the management somewhat easier and cleaner.

Below are some simple instructions for setting up two clusters. One cluster that'll run a single node, and another that will run as many as you wish. This is a basic demonstration of our k8s capabilities.

1. Scaffold and deploy two new clusters.

```
rake "cluster:scaffold[sifchain-1,aws]"
rake "cluster:deploy[sifchain-1,aws]"
rake "cluster:scaffold[sifchain-2,aws]"
rake "cluster:deploy[sifchain-2,aws]"
```

2. Generate as many mnemonic phrases as you want, by running:

```
rake keys:generate:mnemonic 
```

A mnemonic ensures that the keys can be restored easily.

3. Import the generated mnemonics into your local keyring by running:

```
rake "keys:import[<moniker>]" 
```

where `<moniker>` is the name of the node that you'll use the key for (this is important, as you'll need it when you stake). Run the command again for each of the phrases you generated. 

4. Setup a seed node:

```
rake "cluster:sifnode:deploy:standalone[sifchain-1,aws,<namespace>,<image>,<tag>,<moniker>,<mnemonic>]"
```

this will deploy a seed node/validator onto the first cluster. The chain ID, in this example, will be `sifchain-1` _(**we may want to have an additional argument, for the cluster name at some stage, to divorce it from the chain ID**)_.

5. Fund the other nodes.

Whilst no additional node have been created yet, you can still fund their addresses (excluding the seed node - as this is where we'll be sending the funds **from**!). These are just the addresses of the keys you imported above (using the mnemonic):

```
rake "faucet:validator:send[sifchain-1,<from_address>,<to_address>,<amount>,<seed_rpc>]"
```

Where:

```
<from_address> = the address of the seed node you created earlier (you can get this from your local keyring)
<to_address> = the address to send funds to (you can also get this from your local keyring)
<amount> = the amount to send with the token denomination (e.g.: 10000000rowan)
<seed_rpc> = the seed node RPC port (e.g.: tcp://1.2.3.4:26657)
```

6. Create additional nodes:

```
rake "cluster:sifnode:deploy:peer[sifchain-2,aws,<namespace>,<image>,<tag>,<moniker>,<mnemonic>,<peer_address>,<genesis_url>]"
```

again, the moniker should match one of the keys you imported earlier.

7. Obtain the public keys for the nodes.

The public keys are required in order to stake/bond:

```
rake "validator:expose:pub_key[sifchain-2,aws,<moniker>]"
```

Run the command again for each of the phrases you generated.

8. Stake to become an actual validator:

```
rake "validator:stake[sifchain-1,<moniker>,<amount>,<public_key>,<seed_rpc>]"
```

Run the command again for each of the phrases you generated.

### CODE REVIEW

File: deploy/helm/block-explorer/values.yaml [hunk 1]
```diff
@@ -1,89 +1,53 @@
...
-  size: 10Gi
 
 mongodb:
   install: true
   mongodbExtraFlags: "--storageEngine=wiredTiger"
   usePassword: false
   persistence:
-    enabled: true
-    storageClass: ebs-sc
+    enabled: false
```
Comment (ilchebedelovski) [inline deploy/helm/block-explorer/values.yaml:40]: @intl-man we don't need persistence for mongodb as well?

File: deploy/helm/sifnode/templates/ebrelayer-deployment.yaml [hunk 2]
```diff
@@ -65,8 +65,6 @@ spec:
             mountPath: /root
       volumes:
       - name: root
```
Comment (ilchebedelovski) [inline deploy/helm/sifnode/templates/ebrelayer-deployment.yaml:67]: @intl-man if we want to remove the claim we need to remove the name of the claim volume as well.

File: deploy/helm/sifnode/templates/sifnode-deployment.yaml [hunk 3]
```diff
@@ -71,8 +71,6 @@ spec:
             mountPath: /root
       volumes:
       - name: root
```
Comment (ilchebedelovski) [inline deploy/helm/sifnode/templates/sifnode-deployment.yaml:73]: @intl-man if we want to remove the claim we need to remove the name of the claim volume as well.

File: deploy/helm/sifnode/templates/sifnodecli-deployment.yaml [hunk 4]
```diff
@@ -50,5 +50,3 @@ spec:
             mountPath: /root
       volumes:
       - name: root
```
Comment (ilchebedelovski) [inline deploy/helm/sifnode/templates/sifnodecli-deployment.yaml:52]: @intl-man we need to remove the name of the claim volume here as well.

**Review by ilchebedelovski — CHANGES_REQUESTED** [review]

File: deploy/helm/sifnode/templates/sifnode-pvc.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    requests:
+      storage: {{ .Values.persistence.size | quote }}
+  {{- if .Values.persistence.storageClass }}
+  {{- if (eq "-" .Values.persistence.storageClass) }}
+storageClassName: ""
+  {{- else }}
+storageClassName: "{{ .Values.persistence.storageClass }}"
+  {{- end }}
+  {{- end }}
+  {{- end -}}
```
Comment (ilchebedelovski) [inline deploy/helm/sifnode/templates/sifnode-pvc.yml:21]: @intl-man can we maybe just intend correctly the if statements, or maybe that is how GitHub displays the code?

**Review by ilchebedelovski — COMMENTED** [review]
@intl-man I saw that you delete the EBS and EFS creation from Terraform, but still, there is a possibility for enabling persistence in the Hem Charts. What kind of persistence we are using now (just a question)?

**Review by utx0 — APPROVED** [review]
Very nice work here bro!

File: deploy/helm/sifnode/templates/sifnode-pvc.yml (hunk 5 above)
Comment (intl-man) [reply deploy/helm/sifnode/templates/sifnode-pvc.yml]: @ilchebedelovski Done.

**Comment by intl-man** [comment]
> @intl-man I saw that you delete the EBS and EFS creation from Terraform, but still, there is a possibility for enabling persistence in the Hem Charts. What kind of persistence we are using now (just a question)?

Have a look at the `helm/sifnode/templates/sifnode-pvc.yml` and in the main deployment YAML. The setup is very simple - terraform not required!

**Pull request merged**